r&d.md

```diff
@@ -147,52 +146,94 @@
     ? call it `:define="x, y, z"`?
       -> it seems `:with="x=1, y=2"` works well. `:let` has dissonance with js'y let.
     ? how to extend state
-  3. `:with.x="1", :with.y="2"`
-    + easier to parse, since init code can be _messy_
-    - against paradigm
+    + It already just works via `:fx="x=1, y=2"` since we do sandboxing...
+
+### [x] `:scope="a=1,b=2"` instead of with="{...}"? -> yes, but better do both
+
+  + shorter syntax
+  + on par with django, liquid
+  ~ avoids js with association
+  + enables per-variable effects
+  + it's more natural for immediate scope to have access to vars `:fx="x=1,y=2,get=()=>x++"`
+    * rather than creating a layer of storage
+  - unclear/complicated parsing
+  ? or `:define="a=1, b=2"`, `:let="a=1"`
+  - can simply be done via `:fx="a=1, b=2"`
+
+### [x] What's the best name for :scope/:with/:data? -> :scope, but :with is different directive.
+
+  1. :with
+    - bad remembrance of JS with
+      +~ not necessarily the case
+    - `:with="{x:1,y:2,get(){x+y}}"` has problem of immediate scope access, we need define variables instead
+    - `:with` looks weird although it's supposed to make sense
+  2. :scope
+    + the most direct name for "block scope"
+    + `js-scope` makes sense
+    + can be used without data, just to indicate a separate scope
+    + on par with petit-vue
+    + allows autoinit only :scope parts
+    - scope attribute is used for th
+      ~ rare, but can be set via `:={scope:'xyz'}`
+    - :scope/@scope is used in CSS
+  3. :=
+    - cannot be used on its own
+  4. :local=""
+  5. :ctx=""
```

Comment:
scope is fine, reminds me a bit of old AngularJS though, `ctx` is nice too, but has a "middleware'ish" sound to me, but it would be applicable, guess `:bind` would not be appropriate either?

Reply:
There's `v-bind` in vue and `x-bind` in alpine, so it would disrupt convention. `:scope` seems better than `:with` because it can create a scope without any new variables, just simply `<div :scope><span :fx="x=1"></span></div>` - so that `x` writes to `scope`, not global. Besides `:scope` can return non-objects, like `:scope="x=1,y=2"` - which just defines variables, unlike `:with` that would expect an object. Essentially `:with` has same disadvantage as _with_ in JS.